src/main.rs

```diff
@@ -1,23 +1,85 @@
+use std::time::Duration;
+
 use macroquad::prelude::*;
 
 use conways::cell::Cell;
 use conways::grid::Grid;
 
 #[macroquad::main("Conway's Game of Life")]
 async fn main() {
-    clear_background(BLACK);
-
-    let mut grid = Grid::new_from_cells(vec![
-        vec![Cell::Dead, Cell::Alive, Cell::Dead, Cell::Dead],
-        vec![Cell::Alive, Cell::Alive, Cell::Dead, Cell::Dead],
-        vec![Cell::Dead, Cell::Dead, Cell::Alive, Cell::Dead],
-        vec![Cell::Dead, Cell::Dead, Cell::Dead, Cell::Alive],
-    ]); // This is an example grid, for testing purposes only
+    let (rows, cols) = (50, 50);
+    let mut grid = Grid::new(rows, cols);
+    let mut paused = true;
+    let mut show_info = true;
 
     loop {
-        // Here I should draw the grid state on screen I guess.
+        // If it is not paused
+        clear_background(BLACK);
+
+        // Width and Height are defined on every tick cause screen can be resized.
+        let scr_w = screen_width();
+        let scr_h = screen_height();
+        let cell_w = scr_w / grid.cols() as f32;
+        let cell_h = scr_h / grid.rows() as f32;
+
+        // Draw Board
+        #[allow(clippy::needless_range_loop)]
+        for i in 0..grid.rows() {
+            for j in 0..grid.cols() {
+                if grid.cells[i][j] == Cell::Alive {
+                    draw_rectangle(cell_w * j as f32, cell_h * i as f32, cell_w, cell_h, WHITE);
+                }
+            }
+            // Draw vertical an horizontal lines of grid
+            draw_line(0.0, cell_h * i as f32, scr_w, cell_h * i as f32, 1.0, GRAY);
+            draw_line(cell_w * i as f32, 0.0, cell_w * i as f32, scr_h, 1.0, GRAY);
+        }
+
+        // Handle KeyPress: R (Random), C (Clear), Space (Play/Pause), I (Toggle Info)
+        match get_last_key_pressed() {
+            Some(KeyCode::R) => grid = Grid::new_random(rows, cols),
+            Some(KeyCode::C) => grid = Grid::new(rows, cols),
+            Some(KeyCode::Space) => paused = !paused,
+            Some(KeyCode::I) => show_info = !show_info,
+            Some(_) | None => {}
```

Comment:
The options are great, I would just add one more, something like `X` to end the game. So you can handle the ending of the program graciously and the user doesn't have to run `ctrl + C`.

Reply:
Great idea! Hadn't thought about that. I've added `Q` to end the game, also updated the info text accordingly. Thanks for the suggestion!